### PR TITLE
Update filament to 1.10.7

### PR DESCRIPTION
**Pull request by RGregat**
### Fri Jul 16 09:59:17 2021 +0200
**Added a flag to test a certain behaviour**
* During an `onUpdate(FrameTime frameTime)` call in `BaseArFragment`
Sceneform is asking ARCore to hand out an AugmentedImageDatabase, even
if it is null. I added for testing purposes a new flag to prevent that.
The default value is `true` to not bother the normal SDK-User with it.
So why did I do this? Well, I found in a Stacktrace the call to
`getAugmentedImageDatabase()`. And on the other hand, why do such a call
if no AugmentedImageDatabase is set.

### Fri Jul 16 09:55:34 2021 +0200
**Updated Filament from 1.10.6 to 1.10.7**
* All `*.mat` files are compiled to `*.filamat` files.
* The `MaterialProvider` is now an interface and can't be used directly.
* A replacement is the `UberShaderLoader`, this is an implementation of
the `MaterialProvider`. See https://github.com/google/filament/pull/4231
for more information.
* Tested the new Filament Version against all samples. Only the
**sample-sceneview-background** sample is not working correctly. I'll
check that. --> **This sample is also not working with the previous version of Filament. But it might also simple be a problem with my Smartphone**
* The following is still needed, if a `ViewRenderable` should show the
correct color.

```java
Renderer renderer = arSceneView.getRenderer();

if (renderer != null) {
   renderer.getFilamentView().setColorGrading(
      new ColorGrading.Builder()
         .toneMapping(ColorGrading.ToneMapping.FILMIC)
         .build(EngineInstance.getEngine().getFilamentEngine())
   );
}
```

### CODE REVIEW

**Review by ThomasGorisse — COMMENTED** [review]
Frame AugmentedImagesDatabase checks are here just for a session.config changed behavior after init.
We should only consider the config as configurable at the session start to avoid this.
I propose to have an `updateSessionConfig()` function which is called internally after the first session config and also public to allow possible new session modifications (even if we know that not everything can be changed in the config after init)
The `getAugmentedImagesDatabase()!=null`&& `imageCount > 0` can be done here.

**Comment by ThomasGorisse** [comment]
I checked memory usage and I confirm that the `getArSceneView().getSession().getConfig().getAugmentedImageDatabase()` call is an enormous memory killer.

**Comment by ThomasGorisse** [comment]
@RGregat I made a big clean. Can you make a quick check of the latest master version before I release ?

**Comment by RGregat** [comment]
Ah hey @ThomasGorisse sorry I was busy yesterday, but I will go over the changes when I'm back on my PC.

**Comment by ThomasGorisse** [comment]
Thanks.

I hope you are not underwater with the current weather conditions in Germany.

File: ux/src/main/java/com/google/ar/sceneform/ux/BaseArFragment.java [hunk 1]
```diff
@@ -628,27 +617,26 @@ public void onPeekTouch(HitTestResult hitTestResult, MotionEvent motionEvent) {
...
+
+            // Instructions for the Augmented Image finding mode.
+            boolean showAugmentedImageInstructions = !getArSceneView().isTrackingAugmentedImage();
             if (getInstructionsController().isVisible(InstructionsController.TYPE_AUGMENTED_IMAGE_SCAN) != showAugmentedImageInstructions) {
                 getInstructionsController().setVisible(InstructionsController.TYPE_AUGMENTED_IMAGE_SCAN, showAugmentedImageInstructions);
             }
         }
+
+        for (AugmentedImage augmentedImage : getArSceneView().getUpdatedAugmentedImages()) {
+            onAugmentedImageUpdateListener.onAugmentedImageTrackingUpdate(augmentedImage);
```
Comment (RGregat) [inline ux/src/main/java/com/google/ar/sceneform/ux/BaseArFragment.java:638]: If I'm not overlooked something `onAugmentedImageUpdateListener` could be null. So we need a `null` check or an Optional.

**Comment by RGregat** [comment]
> Thanks.
> 
> I hope you are not underwater with the current weather conditions in Germany.

I live in Berlin and the weather condition is by far better than in west germany. What I see in the daily news is crazy. 3 m heigh waves, completly destroyed houses and so on. Unbelievable. Here in Berlin it is just hot and dry, not much rain, just a few hefty and short thunderstormes. How is it looking in France?
Btw I found only one tiny part which might need a small null check, apart from that I really like the Trackables class.

**Pull request merged**